rdr_service/tools/tool_libs/curation.py

```diff
@@ -372,13 +376,21 @@ def _get_base_src_clean_answers_select(session):
                 QuestionnaireResponseAnswer.valueString.isnot(None)
             ),
             QuestionnaireResponse.status != QuestionnaireResponseStatus.IN_PROGRESS,
-            QuestionnaireResponse.isDuplicate.is_(False)
+            QuestionnaireResponse.isDuplicate.is_(False),
+            or_(
+                ParticipantSummary.dateOfBirth.is_(None),
+                and_(
+                    ParticipantSummary.dateOfBirth.isnot(None),
+                    func.year(func.from_days(func.datediff(ParticipantSummary.consentForStudyEnrollmentAuthored,
```

Comment:
Thanks for the comments :)
1. The key for this is https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_from-days
2. This is a good question, I will ask the requester about this.

Reply:
@wangy70 the original issue was flagged by curation as the actual DOB being > 18yrs, there was not correlation of consentForStudyEnrollmentFirstYesAuthored in the dataset.

Reply:
Maybe I can directly use TIMESTAMPDIFF(YEAR, date_of_birth,consent_date), let me try.

Reply:
@wangy70 It sounds from the Jira conversation as if this calculation should be based on the now() date at the time the code is run (real-time age calcuation) rather than the age at consent based on consent authored date.

Reply:
Also if it helps, curation was using `where date_diff(DATE '2021-04-01', DATE(birth_datetime), YEAR) < 18`, guess the report was run in May for the April data.

Reply:
Thanks @dtharpeuno ! It's much clear now, I just updated the ticket description to reflect this. I will update the code.

Reply:
Done, please review.